File: Sample/nodejs/src/Int3.js

const Int2 = require('./Int2').Int2

class Int3 {
    
    /*   默认值() */
    getvalue1() { return this._value1; }
    ID() { return this._value1; }
    /*   默认值() */
    getvalue2() { return this._value2; }
    
    GetData(key) {
        if ("value1" == key) return this._value1;
        if ("value2" == key) return this._value2;
        return null;
    }
    
    IsInvalid() { return false; }
    
    Set(value) {
        this._value1 = value._value1;
        this._value2 = value._value2;
    }
    
    ToString() {
        return `{
            value1 : ${_value1}, 
            value2 : ${_value2}
            }`;
    }
    
    static Read(fileName, reader) {
        let ret = new Int3();
        ret._value1 = Int2.Read(fileName, reader);
        ret._value2 = reader.ReadInt32();
        return ret;
    }
}
exports.Int3 = Int3;

File: Sample/CSharp/Code/Int3.cs
//本文件为自动生成，请不要手动修改
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using ScorpioProto.Commons;
using ScorpioProto.Table;

namespace Datas {
public partial class Int3 : IData {
    
    private ReadOnlyCollection<Int2> _Value1;
    /* <summary>   默认值() </summary> */
    public ReadOnlyCollection<Int2> getValue1() { return _Value1; }
    private int _Value2;
    /* <summary>   默认值() </summary> */
    public int getValue2() { return _Value2; }
    
    public object GetData(string key) {
        if ("Value1".Equals(key)) return _Value1;
        if ("Value2".Equals(key)) return _Value2;
        return null;
    }
    
    public static Int3 Read(string fileName, IScorpioReader reader) {
        var ret = new Int3();
        {
            var list = new List<Int2>();
            var number = reader.ReadInt32();
            for (int i = 0; i < number; ++i) { list.Add(Int2.Read(fileName, reader)); }
            ret._Value1 = list.AsReadOnly();
        }
        ret._Value2 = reader.ReadInt32();
        return ret;
    }
    
    public void Set(Int3 value) {
        this._Value1 = value._Value1;
        this._Value2 = value._Value2;
    }
    
    public override string ToString() {
        return $"Value1:{_Value1}, Value2:{_Value2}, ";
    }
}
}
